content/en/docs/ops/configuration/extensibility/wasm-module-distribution/index.md

```diff
@@ -0,0 +1,148 @@
+---
+title: Distribute Remote WebAssembly Module
+description: .
+weight: 10
+aliases:
+  - /help/ops/extensibility/distribute-remote-wasm-module
+  - /docs/ops/extensibility/distribute-remote-wasm-module
+keywords: [extensibility,Wasm,WebAssembly]
+owner: istio/wg-policies-and-telemetry-maintainers
+test: n/a
+---
+
+Istio provides the ability to [extend proxy functionality using WebAssembly (Wasm)](/blog/2020/wasm-announce/).
+One of the key advantages of proxy Wasm extensibility is that the extension can be loaded dynamically at runtime.
+Before loading, the Wasm extension needs to be distributed to the proxy.
+Istio provides a way to achieve this by downloading the Wasm module at Istio agent.
+
+## Configure a HTTP Filter with Remote Wasm Module
+
+Here we will walk through an example of adding a basic auth extension to our mesh. We will configure Istio to pull a [basic auth module](https://github.com/istio-ecosystem/wasm-extensions/tree/master/extensions/basic_auth) from a remote URI and load it with configuration to protect the `/productpage` URL.
+
+To configure a WebAssembly filter with a remote Wasm module, two `EnvoyFilter` resources will be installed: one injects the HTTP filter, and the other one provides configuration for the filter which uses the remote Wasm module.
+
+With the first `EnvoyFilter`, an HTTP filter will be injected into gateway proxies. It is configured to request the extension configuration named as `istio.basic_auth` from `ads` (i.e. Aggregated Discovery Service), which is the same configuration source that Istiod uses to provide all other configuration resources. Along with the configuration source, the initial fetch timeout is also set, in order to prevent filters with slow or failed Wasm module remote fetch becomes effective.
```

Comment:
Last sentence here doesn't make sense, I can't quite figure out what it is trying to say. What is the fetch timeout preventing?

Reply:
I broke the explanation into two parts. First part explains what this option does, and the other part explains why this option is needed below with the second `EnvoyFilter` resource. Hope this could make it more clear.